pulp_python/app/models.py

```diff
@@ -229,28 +241,39 @@ def _build_additions(self, additions, remote_metadata):
                 })
             yield content
 
-    def _build_removals(self, removals):
+    def _build_removals(self, removals, current_version):
         """
         Generate the content to be removed.
 
+        Args:
+            removals (set): of filenames to remove
+            current_version (pulpcore.plugin.models.RepositoryVersion): the latest pre-existing
+                version or None if one does not exist.
         Returns:
             generator: A generator of content to be removed.
         """
         q = models.Q()
         for content_key in removals:
             q |= models.Q(pythonpackagecontent__filename=content_key)
-            q_set = self.repository.content.filter(q)
+            q_set = current_version.content().filter(q)
             q_set = q_set.only('id')
             for content in q_set:
                 yield content
 
-    def sync(self):
+    def sync(self, new_version, current_version):
```

Comment:
Sorry to re-nitpick the semantics... This runs in a task, so by the time this runs, it may not be the latest version anymore. I'm not sure how we handle that in pulpcore, but we shouldn't be concerned with whether it is latest or not here. I think "base_version" would be more correct.